### ISSUE TITLE
RuntimeError：loss.backward(retain_graph=True)

### ISSUE THREAD

**Issue by Bear-kai** [mufengjun260/TemporalFusion21#1]
发生异常: RuntimeError
one of the variables needed for gradient computation has been modified by an inplace operation: [torch.cuda.FloatTensor [21, 128, 1, 1]], which is output 0 of UnsqueezeBackward0, is at version 2; expected version 1 instead. Hint: enable anomaly detection to find the operation that failed to compute its gradient, with torch.autograd.set_detect_anomaly(True).

----

1. 请问作者使用的pytorch版本是？是否遇到过上述error；
2. 目测这个问题的原因是：由于始终调用`loss.backward(retain_graph=True)`，即运算图始终未被释放，然后optimizer.zero_grad()就执行了inplace操作，接着再执行`loss.backward(retain_graph=True)`就会导致上述错误；
3. 再看前向传播 `pred_r, pred_t, pred_c, x_return = estimator(...)`，得到当前帧pred_r的过程中，其实用到了之前帧的网络预测（通过`self.last_R_total`），这个势必要求计算图不能被释放，即要求`retain_graph=True`，就势必会导致第2点中的问题？！

还请作者抽空解答一下，非常感谢！

**Comment by Bear-kai** [comment]
【update】以上bug出现在pytorch1.7.1中，改用pytorch.1.1.0后消失，应该是pytorch对于backward的相关机制做了修改。
